app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
 
   protected
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
   end
+
+  def create_cart
+    session[:cart_id] = ShoppingCart.create.id unless session[:cart_id]
```

Comment:
I would also suggest that you at least try the approach suggested by @diraulo I did that and I would like you to see the results.